### PR TITLE
Add documentation for Rubocop overrides

### PR DESCRIPTION
**Pull request by nschonni**
Taking a pass at documenting the current Rubocop overrides to have a better sense of which have been opted-in and why.

### CODE REVIEW

**Comment by nschonni** [comment]
@ClearlyClaire how about this as a start to making sure the rules are documented? I couldn't tell for some of the overrides if the values were just to make CI happy before vs. explicit choices so I left the `Reason:` blank

File: .rubocop.yml [hunk 1]
```diff
@@ -43,83 +50,115 @@ Layout/LineLength:
...
+# https://docs.rubocop.org/rubocop/cops_metrics.html#metricsperceivedcomplexity
 Metrics/PerceivedComplexity:
   Max: 16 # RuboCop default 8
 
+# Reason: Prevailing style uses numeric status codes, matches RSpec/Rails/HttpStatus
+# https://docs.rubocop.org/rubocop-rails/cops_rails.html#railshttpstatus
 Rails/HttpStatus:
   EnforcedStyle: numeric
 
+# Reason:
```
Comment (ClearlyClaire) [inline .rubocop.yml:115]: Those files are used exclusively in the `tootctl` command-line tool. Proper exit codes are useful. Though there are maybe better ways to do it.

**Comment by ClearlyClaire** [comment]
I'm not sure we need to retroactively document every style choices, though for some rules (e.g. `Rails/IgnoredColumnsAssignment` if we, as we believe we should, disable it) it may be useful.

In general, I would recommend *not* enabling new rules unless their benefit is discussed first. Some rules are very useful to detect code smell, but a lot of purely style ones would just create noise for no obvious reason. For instance, I do not think `Style/SymbolArray` makes things significantly more readable, it clashes with Rails examples, and may be generally less approachable to newcomers.

File: .rubocop.yml (hunk 1 above)
Comment (nschonni) [reply .rubocop.yml:115]:
```suggestion
# Reason: Allowed only in the `tootctl` CLI application code
```

**Comment by nschonni** [comment]
> In general, I would recommend _not_ enabling new rules unless their benefit is discussed first

I think the new rules will bubble up only when there is a Dependabot PR for an update to one of the rubocop gems. Now that the CI is run across all the files, it would fail if there are any new rules that don't match the current code base. I would see that as the good time to either fix the new issues, punt them into the `.rubocop-todo.yml` for later cleanup, or document why they'll never get fixed in the `.rubocop.yml`

**Review by ClearlyClaire — APPROVED** [review]

**Pull request merged**